magento/magento2
# Show customer_grid indexer as green when realtime

### Description
Magento recommends that all indexers be set in `schedule` mode ("Update by Schedule") except for `customer_grid` which should be in `realtime` mode ("Update on Save"). This is documented here in [the devdocs](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/indexing.html#m2devgde-indexing-modes) and the Site-Wide Analysis Tool reports this also.

<details><summary>Screen-shot of Site-Wide Analysis Tool</summary><hr />

![Screenshot_2021-11-06_21-37-48](https://user-images.githubusercontent.com/334786/140624703-8fadae3a-f1a3-4010-81d6-aa963f71f3ca.png)

<hr /></details>

Currently Magento renders all indexers as 'green' when they are in `schedule` mode, including `customer_grid`. This is misleading as the `customer_grid` is not in the recommended state when set as `schedule` and shows in 'red' when in `realtime` mode which is its recommended state.

This pull request makes the colours match the recommended state for each indexer.

### Related Pull Requests

*None*

### Fixed Issues
*None*

### Manual testing scenarios
1. Set all indexers as `realtime` with: `php bin/magento indexer:set-mode realtime`
1. Observe colours for each indexer in admin (System -> Tools -> Index Management)
1. Set all indexers as `schedule` with: `php bin/magento indexer:set-mode schedule`
1. Observe colours for each indexer in admin (System -> Tools -> Index Management)
1. Set the `customer_grid` indexer as `realtime` with: `php bin/magento indexer:set-mode realtime customer_grid`
1. Observe colours for each indexer in admin (System -> Tools -> Index Management)

<details><summary>Screen-shot of indexers all set 'Update on Save'</summary>

![Screenshot_2021-11-06_21-48-58](https://user-images.githubusercontent.com/334786/140624817-85076f00-5070-4db4-845d-01d58ab7e87f.png)

</details>
<details><summary>Screen-shot of indexers all set 'Update by Schedule'</summary>

![Screenshot_2021-11-06_21-50-32](https://user-images.githubusercontent.com/334786/140624830-34652f80-b6ab-41f7-aa8d-0b50e20f1793.png)

</details>
<details><summary>Screen-shot of indexers set to their recommended modes</summary>

![Screenshot_2021-11-06_21-45-49](https://user-images.githubusercontent.com/334786/140624848-09d7addb-0b44-40bf-8d6c-6d27b1ddfdbf.png)

</details>

### Questions or comments
I have made the function which determines the preferred mode public to allow for extensibility with plugins.

Should the Design Config Grid indexer also be in this category of 'prefer realtime mode'?

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#34648: Show customer_grid indexer as green when realtime
